lib/bitcrowd_ecto/changeset.ex

```diff
@@ -112,4 +112,134 @@ defmodule BitcrowdEcto.Changeset do
       add_error(changeset, field, "is not a valid url", validation: :format)
     end
   end
+
+  @doc """
+  Validates a field timestamp to be in the past, if present
+  """
+  @doc since: "0.6.0"
+  @spec validate_past_datetime(Ecto.Changeset.t(), atom, DateTime.t()) :: Ecto.Changeset.t()
+  def validate_past_datetime(changeset, field, now \\ DateTime.utc_now()) do
+    datetime = get_change(changeset, field)
+
+    if datetime && DateTime.compare(now, datetime) == :lt do
+      add_error(changeset, field, "must be in the past", validation: :date_in_past)
+    else
+      changeset
+    end
+  end
+
+  @doc """
+  Validates a field timestamp to be in the future, if present
+  """
+  @doc since: "0.6.0"
+  @spec validate_future_datetime(Ecto.Changeset.t(), atom, DateTime.t()) :: Ecto.Changeset.t()
+  def validate_future_datetime(changeset, field, now \\ DateTime.utc_now()) do
+    datetime = get_change(changeset, field)
+
+    if datetime && DateTime.compare(now, datetime) != :lt do
+      add_error(changeset, field, "must be in the future", validation: :date_in_future)
+    else
+      changeset
+    end
+  end
+
+  @doc """
+  Validates a field timestamp to be after the given one
+  """
+  @doc since: "0.6.0"
+  @spec validate_datetime_after(Ecto.Changeset.t(), atom, DateTime.t(), [{:formatter, fun}]) ::
+          Ecto.Changeset.t()
+  def validate_datetime_after(changeset, field, reference_datetime, opts \\ []) do
+    formatter = Keyword.get(opts, :formatter, &DateTime.to_string/1)
+    datetime = get_change(changeset, field)
+
+    if datetime && DateTime.compare(reference_datetime, datetime) != :lt do
+      add_error(
+        changeset,
+        field,
+        "must be after #{formatter.(reference_datetime)}",
```

Comment:
Sorry for nitpicking again, but I didn't spot this in the previous review: I think this is going through Gettext in the web layer, right? In that case, it should be

```suggestion
        "must be after %{reference}",
        reference: formatter.(reference_datetime),
```

Reply:
👍 In that case I have to drop my tests on passing a custom formatter! Or is there a way to get the error message as interpreted by gettext?

Reply:
Nevermind what I just said 🔪